### PR TITLE
Split QueryCondition component out of QueryBuilder

### PR DESCRIPTION
**Pull request by Ladsgroup**
Bug: T269133

### CODE REVIEW

File: src/components/QueryCondition.vue [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+export default Vue.extend( {
+	name: 'QueryCondition',
+	data() {
+		return {
+			fieldErrors: {
+				property: null as null | Error,
+				value: null as null | Error,
+			},
+			limitedSupport: false,
+			propertyValueRelation: PropertyValueRelation,
```
Comment (micgro42) [inline src/components/QueryCondition.vue]: As far as I can tell, this is unused can just be dropped.

File: src/components/QueryCondition.vue [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+					this.textInputValue = '';
+				}
+				this.$store.dispatch( 'updatePropertyValueRelation', selectedPropertyValueRelation );
+			},
+		},
+		textInputValue: {
+			get(): string { return this.$store.getters.value; },
+			set( value: string ): void { this.$store.dispatch( 'updateValue', value ); },
+		},
+		...mapState( {
```
Comment (micgro42) [inline src/components/QueryCondition.vue]: We're not using this at all in this component

File: src/components/QueryBuilder.vue [hunk 3]
```diff
@@ -10,26 +10,9 @@
...
-					ref="value"
-					v-model="textInputValue"
-					:error="fieldErrors.value ?
-						{message: $i18n(fieldErrors.value.message), type: fieldErrors.value.type}: null"
-					:placeholder="$i18n('query-builder-input-value-placeholder')"
-					:disabled="selectedPropertyValueRelation === propertyValueRelation.Regardless"
-				/>
-			</div>
+			<QueryCondition
+				ref="firstCondition"
```
Comment (micgro42) [inline src/components/QueryBuilder.vue]: Are we using this somewhere?

File: src/components/QueryCondition.vue [hunk 4]
```diff
@@ -0,0 +1,132 @@
+<template>
+	<div class="querybuilder__rule" :fieldErrorsrrr="fieldErrorsrrr">
```
Comment (micgro42) [inline src/components/QueryCondition.vue]: `fieldErrorsrrr` seems unused?

File: src/components/QueryCondition.vue [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+			get(): string { return this.$store.getters.value; },
+			set( value: string ): void { this.$store.dispatch( 'updateValue', value ); },
+		},
+		...mapState( {
+			errors: 'errors',
+		} ),
+	},
+	watch: {
+		'$store.state.errors': function (): void {
+			this.validate();
```
Comment (micgro42) [inline src/components/QueryCondition.vue]: Do we actually need this? What does it do?

**Review by micgro42 — CHANGES_REQUESTED** [review]
Looking good, but I think there are still some things left to finish

File: src/components/QueryCondition.vue (hunk 1 above)
Comment (Ladsgroup) [reply src/components/QueryCondition.vue]: Done

File: src/components/QueryCondition.vue (hunk 2 above)
Comment (Ladsgroup) [reply src/components/QueryCondition.vue]: Done

File: src/components/QueryBuilder.vue (hunk 3 above)
Comment (Ladsgroup) [reply src/components/QueryBuilder.vue]: No, I put it for later but we can add it later too

File: src/components/QueryCondition.vue (hunk 4 above)
Comment (Ladsgroup) [reply src/components/QueryCondition.vue]: Oh it was leftover from debug.

File: src/components/QueryCondition.vue (hunk 5 above)
Comment (Ladsgroup) [reply src/components/QueryCondition.vue]: Yeah we need this. It basically adds the per-field validation messages when the query is incomplete. We should rework it once the state is in a better shape.

File: src/components/QueryCondition.vue (hunk 1 above)
Comment (Ladsgroup) [reply src/components/QueryCondition.vue]: It is used in the template: propertyValueRelation.Regardless, when I removed it, lots of tests failed with:
 ● QueryCondition.vue › updates the store value when the user fills in the value textfield

    TypeError: Cannot read property 'Regardless' of undefined

Comment (micgro42) [reply src/components/QueryCondition.vue]: That should be `PropertyValueRelation.Regardless` (capital `P`, the thing that is imported)

Comment (Ladsgroup) [reply src/components/QueryCondition.vue]: That doesn't work, I think inside template, you can't do it. I can extract it to a method and use that instead.

File: src/components/QueryCondition.vue (hunk 5 above)
Comment (micgro42) [reply src/components/QueryCondition.vue]: I see. That works, but is quite magical. Going with the ref after all is maybe the better idea. I.e. having a ref `condition` on the `QueryCondition` in the `QueryBuilder` component and then calling `condition.validate()` in the `QueryBuilder`'s `validate` method.

You may need the following code for giving the correct type to the ref: https://github.com/wmde/wikit/commit/bad24a9eba4a2be4fec9539d973c6636381bd212#diff-c000023c1d09de1efda5d372b6a61658f3158308cbb6d585822d328ac512389eR55

File: src/components/QueryBuilder.vue [hunk 6]
```diff
@@ -82,31 +51,17 @@ export default Vue.extend( {
 			this.$store.dispatch( 'incrementMetric', metric );
 		},
 		validate(): void {
+			// TODO: Clean up FormValue <-> Query relation
```
Comment (micgro42) [inline src/components/QueryBuilder.vue:55]: Maybe we can transform this into a getter `isQueryBuilderValid`? Then we can look at its result in this method and set the state accordingly. But that can be done in a follow-up

File: src/components/QueryCondition.vue (hunk 1 above)
Comment (micgro42) [reply src/components/QueryCondition.vue]: > I can extract it to a method and use that instead.

Yes, a dedicated Method determining whether the Dropdown is disabled seems less confusing than a state that is actually just used as an enum.

File: src/components/QueryBuilder.vue (hunk 6 above)
Comment (Ladsgroup) [reply src/components/QueryBuilder.vue:55]: Yes, This PR is already getting pretty big.

File: src/components/QueryCondition.vue (hunk 1 above)
Comment (Ladsgroup) [reply src/components/QueryCondition.vue]: Done

File: src/components/QueryCondition.vue (hunk 5 above)
Comment (Ladsgroup) [reply src/components/QueryCondition.vue]: Done

Comment (micgro42) [reply src/components/QueryCondition.vue]: Yay :)

File: src/components/QueryCondition.vue (hunk 1 above)
Comment (micgro42) [reply src/components/QueryCondition.vue]: :+1:

**Review by micgro42 — APPROVED** [review]

**Pull request merged**